### PR TITLE
network: use request queue to configure DHCP clients

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by keszybz** [comment]
Needs a rebase now that #19980 has been merged.

**Comment by yuwata** [comment]
Rebased.

File: src/network/networkd-dhcp4.c [hunk 1]
```diff
@@ -1642,17 +1626,83 @@ int dhcp4_update_mac(Link *link) {
...
+
+        if (!link_dhcp4_enabled(link))
+                return 0;
+
+        if (link->dhcp_client)
+                return 0;
+
+        r = link_queue_request(link, REQUEST_TYPE_DHCP4_CLIENT, NULL, false, NULL, NULL, NULL);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to request to configure DHCP4 client: %m");
```
Comment (keszybz) [inline src/network/networkd-dhcp4.c]: "Failed to request configuring of the DHCPv4 client: %m"

("to request to configure" is ungrammatical. "to request configuration" is OK, but ambiguous: did we request some operation, or did we request the configuration data? "to request configuring" avoids the ambiguity.)

File: src/network/networkd-dhcp4.c [hunk 2]
```diff
@@ -1642,17 +1626,83 @@ int dhcp4_update_mac(Link *link) {
...
+
+        log_link_debug(link, "Requested to configure DHCP4 client.");
```
Comment (keszybz) [inline src/network/networkd-dhcp4.c]: "Requested configuring of the DHCPv4 client."

File: src/network/networkd-dhcp6.c [hunk 3]
```diff
@@ -1716,6 +1716,60 @@ int dhcp6_update_mac(Link *link) {
...
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to configure DHCP6 client: %m");
+
+        r = ndisc_start(link);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to start IPv6 Router Discovery: %m");
+
+        r = dhcp6_start(link);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to start DHCP6 client: %m");
```
Comment (keszybz) [inline src/network/networkd-dhcp6.c]: Hmm, it's DHCP[46] in some places, and DHCPv[46] in other places… We use DHCPv[46] in the docs, and I think we should standardize on that here too.

File: src/network/networkd-dhcp6.c [hunk 4]
```diff
@@ -1716,6 +1716,60 @@ int dhcp6_update_mac(Link *link) {
...
+                return 0;
+
+        if (link->dhcp6_client)
+                return 0;
+
+        r = link_queue_request(link, REQUEST_TYPE_DHCP6_CLIENT, NULL, false, NULL, NULL, NULL);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to request to configure DHCP6 client: %m");
+
+        log_link_debug(link, "Requested to configure DHCP6 client.");
```
Comment (keszybz) [inline src/network/networkd-dhcp6.c]: "Requested configuring of …"

**Review by keszybz — COMMENTED** [review]
Looks good, except for some of the message strings…

**Comment by yuwata** [comment]
@keszybz Thank you for your review. All suggestions are addressed. PTAL.

**Review by keszybz — APPROVED** [review]
LGTM.

**Comment by yuwata** [comment]
focal-amd64 failed with the following, not related.
```
TEST-30-ONCLOCKCHANGE:              FAIL     (154 s)
```

**Pull request merged**